runtime/pkg/pbutil/pbutil.go

```diff
@@ -207,6 +287,66 @@ func ToListValue(v []interface{}, t *runtimev1.Type) (*structpb.ListValue, error
 	return x, nil
 }
 
+// ToListValueUnknown converts a list google.protobuf.List similar to ToListValue but when the type of list is list of unknown type.
+// It uses reflection so should not be used when ToListValue can be used.
+func ToListValueUnknown(s any, t *runtimev1.Type) (*structpb.ListValue, error) {
+	var elemType *runtimev1.Type
+	if t != nil {
+		elemType = t.ArrayElementType
+	}
+	v := reflect.ValueOf(s)
+	x := &structpb.ListValue{Values: make([]*structpb.Value, v.Len())}
+	for i := 0; i < v.Len(); i++ {
+		var err error
+		x.Values[i], err = ToValue(v.Index(i).Interface(), elemType)
```

Comment:
Actually this was written more from the POV of `clickhouse` right now. `nil` is already handled and if `runtimev1.Type` is set as `Array` it is guaranteed to be of type `[]any` but yeah given this gets invoked in a generic context I can add some defensive checks here to throw errors in case expected types are not passed.